python/fusion_engine_client/utils/enum_utils.py

```diff
@@ -7,13 +7,25 @@ class DynamicEnumMeta(EnumMeta):
     UNRECOGNIZED_PREFIX = '_U'
 
     def __call__(cls, value, *args, **kwargs):
+        raise_on_unrecognized = kwargs.pop('raise_on_unrecognized', True)
+
         # If the user passed in a string, redirect the request: (Foo('bar') -> Foo.BAR). Normally, enums use [] for
         # strings and () for integers, but that can lead to confusion.
         if isinstance(value, str):
-            return cls[value]
+            name = value
+            try:
+                return cls[name]
+            except KeyError as e:
+                if raise_on_unrecognized:
+                    raise e from None
+                else:
+                    used_values = {int(v) for v in cls}
+                    unused_value = -1
```

Comment:
I find the intent of this a little confusing. This algorithm appears to assume the order of the values is something like:

[ 1 ,10 ,2, 7, -1, 50, -2, 0, -3, ...] where the order of negative values is assumed, but other values can be in whatever order. If that's the case, it seems like this check would be simpler to be:
`unused_value = min(used_values + [0]) - 1` or maybe even clearer as  `unused_value = min(min(a), 0) - 1`

Reply:
Yeah, I sort of figured negative values are pretty unlikely, but I like your second option better just to be safe.